### PR TITLE
Move Key Vault samples from certificates and secrets to same structure as Keys

### PR DESCRIPTION
**Pull request by vhvb1989**
Updating to follow the same project structure as Key Vault Keys.

The top folder `samples` from the service directory exposes only the md files with the sample's docs.
The sample's code should be inside the test folder.

Also updating/fixing the name for folders, files and cmake-targets

for folders and cmake-targets, we use a `-` to split words
for files, we use snake case

### CODE REVIEW

**Review by gearama — APPROVED** [review]

File: sdk/keyvault/azure-security-keyvault-certificates/README.md [hunk 1]
```diff
@@ -277,22 +277,22 @@ You will notice that additional information is logged, like the client request I
 
 Several Azure Key Vault secrets client library samples are available to you in this GitHub repository. These samples provide example code for additional scenarios commonly encountered while working with Azure Key Vault:
 
-* [Certificate-Basic-Operations](https://github.com/Azure/azure-sdk-for-cpp/tree/main/sdk/keyvault/azure-security-keyvault-certificates/samples/certificate-basic-operations):
```
Comment (ahsonkhan) [inline sdk/keyvault/azure-security-keyvault-certificates/README.md:280]: Why remove the links?

File: sdk/keyvault/azure-security-keyvault-certificates/CMakeLists.txt [hunk 2]
```diff
@@ -87,7 +87,7 @@ endif()
 # endif()
 
 if(BUILD_SAMPLES)
-   add_subdirectory(samples)
+   add_subdirectory(test/samples)
```
Comment (ahsonkhan) [inline sdk/keyvault/azure-security-keyvault-certificates/CMakeLists.txt:90]: Why does it make sense to move these to the `test` directory? Will customers know to be able to find them here?

File: sdk/keyvault/azure-security-keyvault-certificates/test/samples/certificate-get-certificates/CMakeLists.txt [hunk 3]
```diff
@@ -0,0 +1,15 @@
...
+project (certificate-get-certificates LANGUAGES CXX)
+set(CMAKE_CXX_STANDARD 14)
+set(CMAKE_CXX_STANDARD_REQUIRED True)
+
+add_executable (
+  certificate-get-certificates
+  certificate_get_certificates.cpp
+)
+
+target_link_libraries(certificate-get-certificates PRIVATE azure-security-keyvault-certificates azure-identity )
```
Comment (ahsonkhan) [inline sdk/keyvault/azure-security-keyvault-certificates/test/samples/certificate-get-certificates/CMakeLists.txt:15]: nit: remove whitespace
```suggestion
target_link_libraries(certificate-get-certificates PRIVATE azure-security-keyvault-certificates azure-identity)
```

File: sdk/keyvault/azure-security-keyvault-certificates/test/samples/certificate-import-certificate/CMakeLists.txt [hunk 4]
```diff
@@ -0,0 +1,15 @@
...
+project (certificate-import-certificate LANGUAGES CXX)
+set(CMAKE_CXX_STANDARD 14)
+set(CMAKE_CXX_STANDARD_REQUIRED True)
+
+add_executable (
+  certificate-import-certificate
+  certificate_import_certificate.cpp
+)
+
+target_link_libraries(certificate-import-certificate PRIVATE azure-security-keyvault-certificates azure-identity )
```
Comment (ahsonkhan) [inline sdk/keyvault/azure-security-keyvault-certificates/test/samples/certificate-import-certificate/CMakeLists.txt:15]: nit:
```suggestion
target_link_libraries(certificate-import-certificate PRIVATE azure-security-keyvault-certificates azure-identity)
```

File: sdk/keyvault/azure-security-keyvault-certificates/README.md (hunk 1 above)
Comment (vhvb1989) [reply sdk/keyvault/azure-security-keyvault-certificates/README.md:280]: Makes CI to fail on link verification. I will re-enable them after merging. 
See the fail: https://dev.azure.com/azure-sdk/public/_build/results?buildId=1196363&view=logs&j=a129effc-2dd1-54d1-fb5a-ad7bdc0e851d&t=7b4713e3-e110-5049-e231-cc8a1e664c68&l=42

It seams like the link verifier can now check the links that are suppose to be in main within the same branch.  I can make a test updating the link to see if it works in this same PR.

File: sdk/keyvault/azure-security-keyvault-certificates/CMakeLists.txt (hunk 2 above)
Comment (vhvb1989) [reply sdk/keyvault/azure-security-keyvault-certificates/CMakeLists.txt:90]: I believe that's the expected repo structure for C++. The samples/ folder next to the src/ and test/ folders provides only docs and link to the sample's src code inside test/ .  I followed .NET here.
The way I see it, we want to use the samples as test-cases, and it is just a test classification ( Functional testing ).

So, I expect to see all types of tests inside the test/ folder ( unit, functional, performance, stress, etc ).   

see .NET:
https://github.com/Azure/azure-sdk-for-net/tree/main/sdk/keyvault/Azure.Security.KeyVault.Keys/samples

Java structure needs to follow the language convention for packages, it places everything within src/
https://github.com/Azure/azure-sdk-for-java/tree/main/sdk/keyvault/azure-security-keyvault-keys/src

Python and JS uses samples/ folder for doscs and code (not in test/ )
https://github.com/Azure/azure-sdk-for-js/tree/main/sdk/keyvault/keyvault-keys/samples/v4


Storage C++ does not follow any known structure ( no surprise hahaha ).  There's a main launcher app for the samples and each sample is just a function.  This makes hard for customers to do copy-paste, but it is easy to maintain for devs :)

File: sdk/keyvault/azure-security-keyvault-certificates/test/samples/certificate-get-certificates/CMakeLists.txt (hunk 3 above)
Comment (vhvb1989) [reply sdk/keyvault/azure-security-keyvault-certificates/test/samples/certificate-get-certificates/CMakeLists.txt:15]: Not related to this PR, but sure, let's take it

**Pull request merged**